modules/alloy/src/main/ts/ephox/alloy/alien/Boxes.ts

```diff
@@ -1,26 +1,43 @@
 import { window } from '@ephox/dom-globals';
-import { Fun, Struct } from '@ephox/katamari';
+import { Fun } from '@ephox/katamari';
 import { Height, Location, Width, Element, VisualViewport } from '@ephox/sugar';
 
 import { CssPositionAdt } from './CssPosition';
 import * as OuterPosition from '../frame/OuterPosition';
 
-const pointed = Struct.immutable('point', 'width', 'height') as (point: CssPositionAdt, width: number, height: number) => BoxByPoint;
-const rect = Struct.immutable('x', 'y', 'width', 'height');
+const pointed = (point: CssPositionAdt, width: number, height: number): BoxByPoint => ({
+  point: Fun.constant(point),
+  width: Fun.constant(width),
+  height: Fun.constant(height)
+});
+
+export interface Rect {
+  readonly x: () => number;
+  readonly y: () => number;
+  readonly width: () => number;
+  readonly height: () => number;
+}
+
+const rect = (x: number, y: number, width: number, height: number): Rect => ({
+  x: Fun.constant(x),
+  y: Fun.constant(y),
+  width: Fun.constant(width),
+  height: Fun.constant(height)
+});
 
 export interface Bounds {
-  x: () => number;
-  y: () => number;
-  width: () => number;
-  height: () => number;
-  right: () => number;
-  bottom: () => number;
+  readonly x: () => number;
```

Comment:
If you wanted to, all the thunks in `Boxes` could safely be turned into just readonly properties as `Boxes` was only exposed more recently (5.1.0 I think?), so it'll should only be used within alloy/tinymce.

Reply:
Pulling on that thread seems to impact a lot of files, so it's probably a job for later.